app/scripts/controllers/permissions/specifications.js

```diff
@@ -176,6 +188,49 @@ export const getPermissionSpecifications = ({
         }
       },
     },
+
+    [PermissionNames.permittedChains]: {
+      permissionType: PermissionType.Endowment,
+      targetName: PermissionNames.permittedChains,
+      allowedCaveats: [CaveatTypes.restrictNetworkSwitching],
+      subjectTypes: [SubjectType.Website],
+
+      factory: (permissionOptions, requestData) => {
+        if (Array.isArray(permissionOptions.caveats)) {
+          throw new Error(
+            `${PermissionNames.permittedChains}: Received unexpected caveats. Any permitted caveats will be added automatically.`,
+          );
+        }
+
+        if (!requestData.approvedChainIds) {
+          throw new Error(
+            `${PermissionNames.permittedChains} error: No approved networks specified.`,
+          );
```

Comment:
I think every error message in this file has the superfluous "error" in the prefix. They should all be removed:
```suggestion
            `${PermissionNames.permittedChains}: No approved networks specified.`,
          );
```